app/models/spree/gateway/amazon.rb

```diff
@@ -183,6 +183,16 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
+
+      response = @mws.refund(response_code, order.number, payment.credit_allowed, payment.currency)
```

Comment:
But all we are given is response code, how do we know if we have to cancel or refund? The only time cancel is called is when an admin cancels an order. The cancel! method is only called on payments that have been captured, so to refund would be the only option available.

https://github.com/solidusio/solidus/blob/master/core/app/models/spree/order.rb#L762-L769